### ISSUE TITLE
 There was an error connecting to your database. Please try again.

### ISSUE THREAD

**Issue by ufukayyildiz** [nilsramsperger/docker-nodebb#2]
Hi,

after commands it does not create nodebb folder under /etc.

When i try to install on web interface password is blank for redis and a i get below error.

 There was an error connecting to your database. Please try again.

**Comment by nilsramsperger** [comment]
Hello,

I guess you refer to the volume for NodeBB configuration. The directory /etc/nodebb won't be created on your host machine, but inside the container. What you can see on the host is, that calling `docker volume ls` will list a volume called `nodebb-config`. That is where the config is stored.

If you want to start all over with installation, make sure to delete the volumes and the container first:
- `docker stop myNodeBB`
- `docker rm myNodeBB`
- `docker volume rm nodebb-data`
- `docker volume rm nodebb-files`
- `docker volume rm nodebb-config`

It is hard to tell, what is going wrong at the web installer, without knowing what you did there exactly.
Looking at the web installer, there should be three sections:
1) General Instance Setup
Web Address should be the public address on which your NodeBB will be callable. For a test run on your local machine `http://localhost:4567` will be fine. If you run it on a server with a custom URL, it should look someshing like `http://www.yoururl.com:4567`. If you have a reverse proxy in between, use the URL and port on which the proxy is reachable for redirection to the NodeBB.
2) Create an Administrator account
Fill as you like/need
3) Configure your database
Do not change anything there.

**Comment by hunt978** [comment]
i got the same error when i launched this image in my windows docker installation with this `docker-compose.yml`

```yml
version: "2"

services:
  nodebb:
    container_name: "nodebb"
    image: nilsramsperger/nodebb
    ports:
      - "4567:4567"
    volumes:
      - ./nodebb/data:/var/lib/redis
      - ./nodebb/files:/opt/nodebb/public/uploads
      # - ./nodebb/config:/etc/nodebb
    restart: always
```
then i logined in the container and executed following commands:
```
redis-server --port 6379
```
the web installer seemed to respond with and continued it's process. but finally failed totally with `ERR_EMPTY_RESPONSE` reported by the browser.  docker showed that the container just been keep restarting repeatedly.

**Comment by nilsramsperger** [comment]
Can you please check if the container is stuck in the restart loop right after creation without any interaction via the web installer. After that, the `docker logs nodebb` output could provide some insight.

I fear that for some reason the redis server isn't starting properly. To start the container without entering the loop, you could override the default entrypoint. Try `docker run -it --entrypoint /bin/bash nilsramsperger/nodebb`. Then start redis by `/etc/init.d/redis-server start` and check it's logs: `/var/log/redis/redis-server.log`.

With those two logs, we should be able to pinpoint the issue.

**Comment by hunt978** [comment]
Turns out it is the issue brought in by volumes settings.  

When I opt-ed out all the custom volumes in ‘docker-compose.yml’, everything worked as they should be. After the installation finished, I copied everything that suggested will be updated by the container to the host machine. 

Altered ‘docker-compose.yml’ back and fired up the container, a well configured running NodeBB hadn’t showed up beyond a nginx proxy server. 

So I just “777” everything in the custom volume folder on the host machine ( by the way, I switched back to a Linux server).

‘’’
su chmod -R 777 nodebb
‘’’

Restarted the container, everything just begun working smoothly and beautifully.

Last but not least, thanks for your great work!
